libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -349,6 +355,34 @@ func (p *pod) emitEvents(pod *kubernetes.Pod, flag string, containers []kubernet
 				},
 			}
 			p.publish(event)
+			emitted++
+		}
+	}
+
+	// Finally publish a pod level event so that hints that have no exposed ports can get processed.
+	// Log hints would just ignore this event as there is no ${data.container.id}
```

Comment:
Should we always publish one event per container and one event per pod, properly labelling them, so each handler can decide what to do without depend on the presence of `container.id` or `port`?